### ISSUE TITLE
The cloud controller should allow me to filter the nodes it's managing

### ISSUE THREAD

**Issue by christian-schlichtherle** [hetznercloud/hcloud-cloud-controller-manager#753]
### TL;DR

We are running an IoT project using K3s. Cloud nodes are provisioned in the Hetzner cloud, edge nodes at customer premises not (obviously). Unfortunately, the controller manager wants to manage all nodes, not just the cloud nodes in the Hetzner data center. This has unwanted side effects: For example, when an edge node goes offline, the cloud controller manager responds with this:

```
I0928 09:13:19.398476       1 event.go:389] "Event occurred" object="de-be-13353-fd-p0" fieldPath="" kind="Node" apiVersion="" type="Normal" reason="DeletingNode" message="Deleting node de-be-13353-fd-p0 because it does not exist in the cloud provider"
```

This causes the edge node to be completely deleted, thereby also losing its node labels, annotations and taints.

### Expected behavior

All it takes to remedy this issue would be to set some node selector based on node labels so that the cloud controller leaves the nodes not matching the node selector alone.

Is it possible?

**Comment by christian-schlichtherle** [comment]
No worries: I figured I can just patch the nodes with their own, `.spec.providerID`. When it's non empty and doesn't start with hcloud://, hrobot://, hcloud://bm- then the hccm leaves these nodes alone.

**Comment by apricote** [comment]
This should ideally be implemented in https://github.com/kubernetes/cloud-provider/, as all the logic to interact with Kubernetes is implemented there. HCCM only implements a few small interfaces to talk to the Hetzner Cloud API.

**Comment by christian-schlichtherle** [comment]
For context, setting `.spec.providerID` to something other than `hcloud://`, `hrobot://`, `hcloud://bm-` does cause the HCCM to not delete an edge node when it goes offline. However, the HCCM still complains about these node in its logs. Here's a sample:
